students/BrianNisonger/Lesson5/assignment/test_database.py

```diff
@@ -0,0 +1,177 @@
+
+
+@pytest.fixture
+def setup_db(request):
+    mongo = MongoDBConnection()
+    with mongo:
+        db = mongo.connection.furniture
+        db.product_data.drop()
+        db.customer_data.drop()
+        db.rental_data.drop()
+        import_data("dat", "product_data.csv", "customer_data.csv",
+                    "rental_data.csv")
+
+        def cleanup():
+            db.product_data.drop()
+            db.customer_data.drop()
+            db.rental_data.drop()
+
+        request.addfinalizer(cleanup)
+
+
+@pytest.fixture
+def _show_available_products():
+    product_dict = {
+        'prd1001': {
+            'description': '60 inch TV stand',
+            'product_type': 'livingroom',
+            'quantity_available': 2
+        },
+        'prd1003': {
+            'description': 'Hobart Stand Mixer',
+            'product_type': 'kitchen',
+            'quantity_available': 10
+        }
+    }
+    return product_dict
+
+
+@pytest.fixture
+def _show_rentals():
+    rental_dict = {
+        'user1001': {
+            'name': 'Elisa Miles',
+            'address': '4490 Union Street',
+            'phone_number': '206-922-0882',
+            'email': 'elisa.miles@yahoo.com'
+        },
+        'user1002': {
+            'name': 'Maya Data',
+            'address': '4936 Elliot Avenue',
+            'phone_number': '206-777-1927',
+            'email': 'mdata@uw.edu'
+        }
+    }
+    return rental_dict
+
+
+def test_import_data():
+    added, error = import_data("dat", "product_data.csv", "customer_data.csv",
+                               "rental_data.csv")
+    assert added == (3, 3, 5)
+    assert error == (0, 0, 0)
+
+
+def test_calculate_availability():
+    rental_list = [{
+        "user": 1001,
+        "product_id": "prd1001"
+    }, {
+        "user": 1002,
+        "product_id": "prd1002"
+    }]
+    product_list = [{
+        "product_id": "prd1001",
+        "quantity_available": "10"
+    }, {
+        "product_id": "prd1002",
+        "quantity_available": "2"
+    }]
+    product_calculated = calculate_availability(product_list, rental_list)
+    for product in product_calculated:
+        if product["product_id"] == "prd1001":
+            assert product["quantity_available"] == 9
+
+
+def test_make_product_dict():
+    product_dict_list = product_list = [{
+        "product_id": "prd1001",
+        "quantity_available": 10,
+        "description": "blah",
+        "product_type": "yes",
+        "garbage_field": "yuck"
+    }]
+    assert make_product_dict(product_dict_list) == {
+        'prd1001': {
+            "description": "blah",
+            "product_type": "yes",
+            "quantity_available": 10
+        }
+    }
+    product_dict_list = product_list = [{
+        "product_id": "prd1001",
+        "quantity_available": 0,
+        "description": "blah",
+        "product_type": "yes",
+        "garbage_field": "yuck"
+    }]
+    assert make_product_dict(product_dict_list) == {}
```

Comment:
These should be two separate tests. They are testing separate concepts (what happens with quantity available is more than one vs 0). Even better, use parameterize. https://docs.pytest.org/en/latest/parametrize.html